backend/src/nodes/properties/outputs/numpy_outputs.py

```diff
@@ -14,12 +18,37 @@ def AudioOutput():
     return NumPyOutput("Audio", "Audio")
 
 
-def ImageOutput(
-    label: str = "Image",
-    image_type: expression.ExpressionJson = "Image",
-):
-    """Output a 2D Image NumPy array"""
-    return NumPyOutput(expression.intersect(image_type, "Image"), label)
+class ImageOutput(NumPyOutput):
+    def __init__(
+        self,
+        label: str = "Image",
+        image_type: expression.ExpressionJson = "Image",
+    ):
+        super().__init__(expression.intersect(image_type, "Image"), label)
+
+    def broadcast(self, value: np.ndarray) -> dict:
+        img = value
+        h, w, c = get_h_w_c(img)
+
+        # resize the image, so the preview loads faster and doesn't lag the UI
```

Comment:
Please move this into its own function instead of copy-pasting. This should include at least the resize+base64 part. I'm not sure about the dict.